nixnet/database/_find_object.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import typing  # NOQA: F401
+
+from nixnet import _cconsts
+from nixnet import _errors
+from nixnet import _funcs
+from nixnet import constants
+
+
+def find_object(parent_handle, object_class, object_name):
+    # type: (int, constants.ObjectClass, typing.Text) -> object
```

Comment:
Regarding 1, I had thought about doing it that way. I guess on the plus side, it eliminates an enum the user has to deal with, and encourages familiarity with the actual objects. On the con side, those of us who use IDEs lose the benefit of intellisense for the enum values. I think my preference would be to continue using the enum as I don't really see much benefit to passing in an object instead of an enum. I'm open to either way though.

Reply:
> On the con side, those of us who use IDEs lose the benefit of intellisense for the enum values. 

Completion for `nixnet.constants.ObjectClass` is not much different than completion for `nixnet.database.`

As for other intellisense features, you can improve that by
1. Providing a common base class for all database objects
2. Change the annotations from `type: (int, constants.ObjectClass, typing.Text) -> object` to `type: (int, typing.Type[nixnet.database.SomeBaseClass], typing.Text) -> nixnet.database.SomeBaseClass`

This is similar to the approach we took elsewhere in the API
- Define [`FrameFactory`](https://github.com/ni/nixnet-python/blob/master/nixnet/types.py#L248) have have each frame implement it
- Accept a frame object when [reading](https://github.com/ni/nixnet-python/blob/7f79f5b3a23c5ffda8f965fcd2d03ad2316f8b88/nixnet/_session/frames.py#L97) and [converting](https://github.com/ni/nixnet-python/blob/e9a946b035db388030b13c426a58f01a5afe4de2/nixnet/convert.py#L216)

> I think my preference would be to continue using the enum as I don't really see much benefit to passing in an object instead of an enum.

Advantages
- Minimize vocabulary terms the user of the API has to deal with.  I need a `Cluster`, so let me pass a `Cluster` vs having to deal with an enum that represents `Cluster`.
- Avoid C-isms.  Part of the role of the enum is a form of RTTI in C.  Changing it to the actual python class is exposing the same concept in the python way
- Controlled access.  The doc string and intellisense will help guide people to the correct values to pass in rather than having things like `System` be an option.